mule-wrapper-additional-parameters-parser/src/main/java/org/mule/runtime/params/AdditionalJvmParameters.java

```diff
@@ -264,10 +267,18 @@ protected static void writeWrapperProps(String[] args, FileWriter writer) throws
     }
   }
 
-  protected static void processBootstrapProperties(Properties bootstrapProperties, FileWriter writer) throws IOException {
+  protected static void processBootstrapProperties(Properties bootstrapProperties, Writer writer) throws IOException {
+    Pattern additionalPattern = compile("wrapper\\.java\\.additional\\.(<n\\d>)(\\.\\w+)?");
+    // Correspond <n> component number with the index in order to re-use the same index for the same component number
+    Map<String, Integer> nComponentIndexMap = new HashMap<>();
+
     for (Entry entry : bootstrapProperties.entrySet()) {
-      if (entry.getKey().toString().matches("wrapper\\.java\\.additional\\.<n\\d>")) {
-        writer.write(wrapperPrefix + ++paramIndex + "=" + entry.getValue().toString() + "\n");
+      Matcher additionalMatcher = additionalPattern.matcher(entry.getKey().toString());
+      if (additionalMatcher.matches()) {
+        String wrapperSuffix = additionalMatcher.group(2) != null ? additionalMatcher.group(2) : "";
+        int index = nComponentIndexMap.computeIfAbsent(additionalMatcher.group(1), k -> ++paramIndex);
+
+        writer.write(wrapperPrefix + index + wrapperSuffix + "=" + entry.getValue().toString() + "\n");
       } else if (entry.getKey().toString().matches("wrapper\\.java\\.classpath\\.<n\\d>")) {
```

Comment:
Not adding a similar `indexMap` for `wrapper.java.classpath` as these properties [don't](https://wrapper.tanukisoftware.com/doc/english/prop-java-classpath-n.html) have a suffix.